src/img2pdf.py

```diff
@@ -320,21 +334,36 @@ def convert(images, dpi=None, pagesize=(None, None), title=None, author=None,
             imgdata = zlib.compress(img)
         im.close()
 
-        # pdf units = 1/72 inch
-        if not pagesize[0] and not pagesize[1]:
-            pdf_x, pdf_y = 72.0*width/float(ndpi[0]), 72.0*height/float(ndpi[1])
-        elif not pagesize[1]:
-            pdf_x, pdf_y = pagesize[0], pagesize[0]*height/float(width)
-        elif not pagesize[0]:
-            pdf_x, pdf_y = pagesize[1]*width/float(height), pagesize[1]
-        else:
-            pdf_x = pagesize[0]
-            pdf_y = pagesize[1]
+        # pdf output based on dpi; point = 1/72 inch
+        pdf_x, pdf_y = 72.0*width/float(ndpi[0]), 72.0*height/float(ndpi[1])
 
         pdf.addimage(color, width, height, imgformat, imgdata, pdf_x, pdf_y)
 
     return pdf.tostring()
 
+def get_ndpi(ndpi, width, height, pagesize, pagesize_options):
+    if pagesize_options['fill'][1]:
+        if width < height:
+            if pagesize[0]:
+                tmp_dpi = 72.0*width/pagesize[0]
+            else:
+                tmp_dpi = 72.0*width/pagesize[1]
```

Comment:
At least one pagesize[0] or pagesize[1] should be assigned.  Will add checking in `valid_size()` to ensure that this is the case.